roles/config-rh-sso/tasks/manage-clients.yml

```diff
@@ -0,0 +1,79 @@
+    - invalid == "protocol"
+
+- name: Set secret value if using client-secret authentication
+  set_fact:
+    secret: "{{ client.client_auth_secret }}" 
+  when: 
+    - client.client_auth_secret is defined
+    - client.client_authenticator_type is defined
+    - client.client_authenticator_type != 'client-secret')
+
+- name: "Create client: {{ client.name | mandatory }} in realm: {{ client.realm | mandatory }}"
+  keycloak_client:
+    auth_client_id: "{{ client.auth_client | default('admin-cli') }}"
+    auth_keycloak_url: "{{ client.auth_url | default('https://' + rh_sso_host + ':' + (rh_sso_port | string)) + '/auth' }}"
+    auth_realm: "{{ client.auth_realm | default('master') }}"
+    auth_username: "{{ client.auth_user | default(sso_admin_user) }}"	
+    auth_password: "{{ client.auth_password | default(sso_admin_pass) }}"
+    state: "{{ client.state | default('present') }}"
+    realm: "{{ client.realm | default(omit) }}"
+    client_id: "{{ client.name | mandatory}}"
+    name: "{{ client.name | mandatory }}"
+    description: "{{ client.description | default(client.name) }}"
+    enabled: "{{ client.enabled | default(true) }}"
+    base_url: "{{ client.base_url | default(omit) }}"
+    admin_url: "{{ client.admin_url | default(omit) }}"
+    root_url: "{{ client.root_url | default(omit) }}"
+    client_authenticator_type: "{{ client.client_auth_type | default(omit) }}"
+    secret: "{{ secret | default(omit) }}"
+    redirect_uris: "{{ client.redirect_uris | default(omit) }}"
+    web_origins: "{{ client.web_origins | default(omit) }}"
+    bearer_only: "{{ client.bearer_only | default(false) }}"
+    consent_required: "{{ client.consent_req | default(false) }}"
+    standard_flow_enabled: "{{ client.standard_flow | default(false) }}"
+    implicit_flow_enabled: "{{ client.implicit_flow | default(false) }}"
+    direct_access_grants_enabled: "{{ client.direct_access_grants | default(false) }}"
+    service_accounts_enabled: "{{ client.svc_acct_enabled | default(false) }}"
+    authorization_services_enabled: "{{ client.auth_svc_enabled | default(false) }}"
+    frontchannel_logout: "{{ client.frontchannel_logout | default(false) }}"
+    protocol: "{{ client.protocol | default(omit) }}"
+    full_scope_allowed: "{{ client.full_scope_allowed | default(false) }}"
+    client_template: "{{ client.template | default('') }}"
+    use_template_config: "{{ client.use_template | default(false) }}"
+    use_template_scope: "{{ client.use_template_scope | default(false) }}"
+    use_template_mappers: "{{ client.use_template_mappers | default(false) }}"
+    surrogate_auth_required: "{{ client.surrogate_auth_req | default(false) }}" 
+    default_roles: "{{ client.default_roles | default(omit)}}"
+    protocol_mappers: "{{ client.protocol_mappers | default(omit) }}"
+    attributes: "{{ client.attributes | default(omit) }}"
+    public_client: "{{ client.public_access | default(false) }}"
+  when:
+    - invalid == ""
```

Comment:
Went with `invalid|trim == ""`

Reply:
oh, sorry if this comment wasn't clear, but I believe you need the full statement (i.e.: the `or` is part of the statement) as otherwise I believe the execution may bail here when it's undefined.